crowbar_framework/app/models/service_object.rb

```diff
@@ -1576,4 +1616,26 @@ def profile(name, &block)
       block.call
     end
   end
+
+  def equal_or_match(value, maybe_regexp)
+    operator = lambda do
+      op, check = maybe_regexp.split(" ", 2)
+      value.to_f.send(op.to_sym, check.to_f)
```

Comment:
So, this is fancy, but actually, if we go the route of integrating operators for comparing versions, I think we should do it right :-)

The `to_f` bit is wrong, and we should instead split on `.` so we can handle versions like 12.04.1 (for Ubuntu), and do a proper comparison (12.04 < 12.04.1).